src/allmydata/test/test_storage_client.py

```diff
@@ -459,6 +460,26 @@ def test_plugin_resource_path(self):
         result = yield do_http(b"get", url)
         self.assertThat(result, Equals(dumps({b"web": b"1"})))
 
+    @inlineCallbacks
+    def test_plugin_resource_persistent_across_requests(self):
+        """
+        The plugin's resource is loaded and then saved and re-used for future
+        requests.
+        """
+        url = u"http://127.0.0.1:{port}/storage-plugins/{plugin_name}/counter".format(
+            port=self.port,
+            plugin_name=self.storage_plugin,
+        ).encode("utf-8")
+        values = {
```

Comment:
Not requesting changes, but I've been trying to use `hyperlink.URL` during poring, because Twisted does and we're using twitsed. Is that worth it?

Reply:
Yea, probably ideal to always handle URLs as structured objects instead of strings of any kind.